### PR TITLE
Add type to lints

### PR DESCRIPTION
**Pull request by kumarde**
Currently, our lints come from the BRs, RFC5280, are ZLint specific, or in some cases, AWSLabs cablint. Adding a field in each lint to keep track of them.

### CODE REVIEW

File: lints/base.go [hunk 1]
```diff
@@ -42,6 +42,17 @@ type LintInterface interface {
 	Execute(c *x509.Certificate) *LintResult
 }
 
+// An Enum to programmatically represent the source of a lint
+type LintType int
```
Comment (dadrian) [inline lints/base.go]: This isn't a `Type` it's a Source. But we set `Source` to a more detailed string, so maybe this is a category? Alternatively, rename `Source` to `ReadableSource` and and then change this to source. @zakird, thoughts?

File: lints/base.go [hunk 2]
```diff
@@ -42,6 +42,17 @@ type LintInterface interface {
...
+
+const (
+	ZeroValue LintType = iota
+	CABFBaselineRequirements
```
Comment (dadrian) [inline lints/base.go]: I'd rename this to `UnknownLintType`, except with whatever we rename `LintType` to.

File: lints/base.go (hunk 1 above)
Comment (kumarde) [reply lints/base.go]: We talked earlier today, "Type" was what we came up with. But I'm okay renaming `Source` to `ReadableSource`, and calling this `Source`.

**Comment by zakird** [comment]
I think this should probably be source and citation now that I've had time
to think about this more.

**Comment by kumarde** [comment]
Got it.

**Comment by kumarde** [comment]
@dadrian @zakird updated.

File: lints/base_test.go [hunk 3]
```diff
@@ -27,12 +27,20 @@ func TestAllLintsHaveNameDescriptionSource(t *testing.T) {
 		if lint.Description == "" {
 			t.Errorf("lint %s has empty description", name)
 		}
-		if lint.Source == "" {
+		if lint.Citation == "" {
 			t.Errorf("lint %s has empty source", name)
```
Comment (zakird) [inline lints/base_test.go]: bad error message

File: lints/base_test.go [hunk 4]
```diff
@@ -27,12 +27,20 @@ func TestAllLintsHaveNameDescriptionSource(t *testing.T) {
...
 		}
 	}
 }
 
+func TestAllLintsHaveType(t *testing.T) {
```
Comment (zakird) [inline lints/base_test.go]: Lints don't have types

**Comment by dadrian** [comment]
Names are fine now.

**Comment by kumarde** [comment]
@dadrian I think I will remove this from the JSON, since Citation has the string already.

File: lints/base.go [hunk 5]
```diff
@@ -56,7 +67,10 @@ type Lint struct {
 	Description string `json:"description,omitempty"`
 
 	// The source of the check, e.g. "BRs: 6.1.6" or "RFC 5280: 4.1.2.6".
-	Source string `json:"source,omitempty"`
+	Citation string `json:"citation,omitempty"`
+
+	// Programmatic source of the check, BRs, RFC5280, or ZLint
+	Source LintSource
```
Comment (dadrian) [inline lints/base.go]: Need to add `json:"-"`

**Pull request merged**